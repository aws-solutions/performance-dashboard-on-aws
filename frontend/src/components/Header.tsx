import React, { ReactNode } from "react";
import { useTranslation } from "react-i18next";

interface Props {
  children: ReactNode;
  className?: string;
}

function Header(props: Props) {
  const { t } = useTranslation();

  return (
    <header
      role="contentinfo"
      aria-label={t("ARIA.Header")}
      className={`border-base-lighter border-width-1px shadow-2 z-top position-sticky top-0 bg-white ${props.className}`}
    >
      {props.children}
    </header>
  );
}

export default Header;
