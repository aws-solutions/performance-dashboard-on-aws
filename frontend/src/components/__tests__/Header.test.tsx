/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Header from "../Header";

test("renders a header", async () => {
    const wrapper = render(
        <Header className="bg-red">
            <p>Test</p>
        </Header>,
    );
    expect(wrapper.container).toMatchSnapshot();
});
