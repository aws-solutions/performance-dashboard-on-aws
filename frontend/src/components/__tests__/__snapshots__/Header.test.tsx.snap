// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a header 1`] = `
<div>
  <header
    aria-label="Header"
    class="border-base-lighter border-width-1px shadow-2 z-top position-sticky top-0 bg-white bg-red"
    role="contentinfo"
  >
    <p>
      Test
    </p>
  </header>
</div>
`;
